src/OICodeGen.cpp

```diff
@@ -986,6 +989,99 @@ std::string OICodeGen::typeToName(drgn_type *type) {
   return typeName;
 }
 
+bool OICodeGen::recordChildren(drgn_type *type) {
+  drgn_type_template_parameter *parents = drgn_type_parents(type);
+
+  for (size_t i = 0; i < drgn_type_num_parents(type); i++) {
+    struct drgn_qualified_type t {};
+
+    if (auto *err = drgn_template_parameter_type(&parents[i], &t)) {
+      LOG(ERROR) << "Error when looking up parent class for type " << type
+                 << " err " << err->code << " " << err->message;
+      drgn_error_destroy(err);
+      continue;
+    }
+
+    drgn_type *parent = drgnUnderlyingType(t.type);
+    if (!isDrgnSizeComplete(parent)) {
+      continue;
+    }
+
+    const char *parentName = drgn_type_tag(parent);
+    if (!parentName) {
+      continue;
+    }
+
+    // drgn pointers are not stable, so use string representation for reverse
```

Comment:
For multiline comments can we use a C style comment please? Exactly in the style of the one currently at line 1026 below.